packages/next/shared/lib/router/router.ts

```diff
@@ -1812,6 +1812,7 @@ export default class Router implements BaseRouter {
             : await fetchNextData({
                 dataHref: this.pageLoader.getDataHref({
                   href: formatWithValidation({ pathname, query }),
+                  skipInterpolation: true,
```

Comment:
This seemed like the most minimal change in order to match the generated cacheKey from https://github.com/vercel/next.js/blob/2633c1ef7a0de8b4b40d3f19aae5aae4f1a5d7cc/packages/next/shared/lib/router/router.ts#L1716-L1721

I'm unsure why this `fetchNextData` doesn't just outright use the `fetchNextDataParams` defined previously though? The key differences between the two currently are this `skipInterpolation` and `hasMiddleware: true`.